--- 001.py	2017-05-09 09:52:49.267019855 +1200
+++ 002.py	2017-05-09 09:53:14.698020194 +1200
@@ -1,5 +1,6 @@
 #!/usr/bin/env python3
 # draw a screen
+# add a player and player control
 
 # GNU All-Permissive License
 # Copying and distribution of this file, with or without modification,
@@ -15,11 +16,27 @@
 Objects
 '''
 
-# put Python classes and functions here
+class Player(pygame.sprite.Sprite):
+    '''
+    Spawn a player
+    '''
+    def __init__(self):
+        pygame.sprite.Sprite.__init__(self)
+        self.images = []
+        for i in range(1,9):
+            img = pygame.image.load(os.path.join('images','hero' + str(i) + '.png')).convert()
+            img.convert_alpha()
+            img.set_colorkey(alpha)
+            self.images.append(img)
+            self.image = self.images[0]
+            self.rect  = self.image.get_rect()
 
 '''
 Setup
 '''
+alpha = (0,0,0)
+black = (0,0,0)
+white = (255,255,255)
 screenX = 960
 screenY = 720
 
@@ -32,6 +49,11 @@
 screen = pygame.display.set_mode([screenX,screenY])
 backdrop = pygame.image.load(os.path.join('images','stage.png')).convert()
 backdropRect = screen.get_rect()
+player = Player()   # spawn player
+player.rect.x = 0
+player.rect.y = 0
+movingsprites = pygame.sprite.Group()
+movingsprites.add(player)
 
 '''
 Main loop
@@ -50,5 +72,6 @@
 
 #    screen.fill(black)
     screen.blit(backdrop, backdropRect)
+    movingsprites.draw(screen) #refresh player position
     pygame.display.flip()
     clock.tick(fps)
