--- 002.py	2017-05-09 09:53:14.698020194 +1200
+++ 003.py	2017-05-09 09:53:47.980020638 +1200
@@ -1,6 +1,7 @@
 #!/usr/bin/env python3
 # draw a screen
 # add a player and player control
+# add player movement
 
 # GNU All-Permissive License
 # Copying and distribution of this file, with or without modification,
@@ -22,6 +23,9 @@
     '''
     def __init__(self):
         pygame.sprite.Sprite.__init__(self)
+        self.momentumX = 0
+        self.momentumY = 0
+        self.frame = 0
         self.images = []
         for i in range(1,9):
             img = pygame.image.load(os.path.join('images','hero' + str(i) + '.png')).convert()
@@ -31,6 +35,41 @@
             self.image = self.images[0]
             self.rect  = self.image.get_rect()
 
+    def control(self,x,y):
+        '''
+        control player movement
+        '''
+        self.momentumX += x
+        self.momentumY += y
+
+    def update(self):
+        '''
+        Update sprite position
+        '''
+
+        currentX = self.rect.x
+        nextX = currentX+self.momentumX
+        self.rect.x = nextX
+
+        currentY = self.rect.y
+        nextY = currentY+self.momentumY
+        self.rect.y = nextY
+
+        # moving left
+        if self.momentumX < 0:
+            self.frame += 1
+            if self.frame > 3*afps:
+                self.frame = 0
+            self.image = self.images[self.frame//afps]
+
+        # moving right
+        if self.momentumX > 0:
+            self.frame += 1
+            if self.frame > 3*afps:
+                self.frame = 0
+            self.image = self.images[self.frame//afps+4]
+
+
 '''
 Setup
 '''
@@ -54,6 +93,7 @@
 player.rect.y = 0
 movingsprites = pygame.sprite.Group()
 movingsprites.add(player)
+movesteps = 10      # how fast to move
 
 '''
 Main loop
@@ -64,7 +104,19 @@
             pygame.quit(); sys.exit()
             main = False
 
+        if event.type == pygame.KEYDOWN:
+            if event.key == pygame.K_LEFT or event.key == ord('a'):
+                player.control(-movesteps,0)
+            if event.key == pygame.K_RIGHT or event.key == ord('d'):
+                player.control(movesteps,0)
+            if event.key == pygame.K_UP or event.key == ord('w'):
+                print('jump')
+
         if event.type == pygame.KEYUP:
+            if event.key == pygame.K_LEFT or event.key == ord('a'):
+                player.control(movesteps,0)
+            if event.key == pygame.K_RIGHT or event.key == ord('d'):
+                player.control(-movesteps,0)
             if event.key == ord('q'):
                 pygame.quit()
                 sys.exit()
@@ -72,6 +124,7 @@
 
 #    screen.fill(black)
     screen.blit(backdrop, backdropRect)
+    player.update()
     movingsprites.draw(screen) #refresh player position
     pygame.display.flip()
     clock.tick(fps)
