--- 003.py	2017-05-09 09:53:47.980020638 +1200
+++ 004.py	2017-05-09 09:54:01.129020813 +1200
@@ -2,6 +2,7 @@
 # draw a screen
 # add a player and player control
 # add player movement
+# add platforms
 
 # GNU All-Permissive License
 # Copying and distribution of this file, with or without modification,
@@ -16,6 +17,23 @@
 '''
 Objects
 '''
+class Platform(pygame.sprite.Sprite):
+    '''
+    Create a platform
+    '''
+    # x location, y location, image width, image height, image file    
+    def __init__(self,xloc,yloc,imgw,imgh,img):
+        pygame.sprite.Sprite.__init__(self)
+        self.image = pygame.Surface([imgw,imgh])
+        self.image.convert_alpha()
+        self.image.set_colorkey(alpha)
+        self.blockpic = pygame.image.load(img).convert()
+        self.rect = self.image.get_rect()
+        self.rect.y = yloc
+        self.rect.x = xloc
+        
+        # paint the image into the blocks
+        self.image.blit(self.blockpic,(0,0),(0,0,imgw,imgh))
 
 class Player(pygame.sprite.Sprite):
     '''
@@ -69,6 +87,17 @@
                 self.frame = 0
             self.image = self.images[self.frame//afps+4]
 
+# map where the platforms go
+# x location, y location, image width, image height, image file
+def level1():
+    platform_list = pygame.sprite.Group()
+    block = Platform(0,591,1920,129, os.path.join('images','block0.png'))
+    platform_list.add(block)
+    block = Platform(20,427,173,72, os.path.join('images','block1.png'))
+    platform_list.add(block)
+    block = Platform(60,227,337,72, os.path.join('images','block2.png'))
+    platform_list.add(block)
+    return platform_list
 
 '''
 Setup
@@ -88,6 +117,7 @@
 screen = pygame.display.set_mode([screenX,screenY])
 backdrop = pygame.image.load(os.path.join('images','stage.png')).convert()
 backdropRect = screen.get_rect()
+platform_list = level1()
 player = Player()   # spawn player
 player.rect.x = 0
 player.rect.y = 0
@@ -124,6 +154,7 @@
 
 #    screen.fill(black)
     screen.blit(backdrop, backdropRect)
+    platform_list.draw(screen)
     player.update()
     movingsprites.draw(screen) #refresh player position
     pygame.display.flip()
