--- 009.py	2017-05-09 23:08:24.226656673 +1200
+++ 010.py	2017-05-11 23:43:46.081061194 +1200
@@ -8,6 +8,7 @@
 # add jump
 # add scrolling
 # add loot
+# add auto enemy
 
 # GNU All-Permissive License
 # Copying and distribution of this file, with or without modification,
@@ -22,6 +23,34 @@
 '''
 Objects
 '''
+class Enemy(pygame.sprite.Sprite):
+    '''
+    Spawn an enemy
+    '''
+    def __init__(self,x,y,img):
+        pygame.sprite.Sprite.__init__(self)
+        self.image = pygame.image.load(os.path.join('images',img))
+        self.image.convert_alpha()
+        self.image.set_colorkey(alpha)
+        self.rect = self.image.get_rect()
+        self.rect.x = x
+        self.rect.y = y
+        self.counter = 0
+        
+    def move(self):
+        '''
+        enemy movement
+        '''
+        if self.counter >= 0 and self.counter <= 100:
+            self.rect.x  += 10
+            self.counter += 1
+        elif self.counter >= 100 and self.counter < 150:
+            self.rect.x  -= 10
+        else:
+            self.counter = 0
+
+        self.counter += 1
+            
 class Platform(pygame.sprite.Sprite):
     '''
     Create a platform
@@ -50,6 +79,7 @@
         self.momentumY = 0
         self.frame     = 0
         self.score     = 0
+        self.health    = 10
         # gravity variables
         self.collide_delta = 0
         self.jump_delta    = 6
@@ -127,6 +157,11 @@
             self.collide_delta += 6
             self.jump_delta    += 6
 
+        hit_list = pygame.sprite.spritecollide(self, enemies, False)
+        for enemy in hit_list:
+            self.health -= 1
+            print(self.health)
+
     def jump(self,platform_list):
         self.jump_delta = 0
 
@@ -180,6 +215,11 @@
 player.rect.y = 0
 movingsprites = pygame.sprite.Group()
 movingsprites.add(player)
+
+enemy   = Enemy(777,531,'owl.png')  # spawn enemy
+enemies = pygame.sprite.Group()
+enemies.add(enemy)
+
 movesteps = 10      # how fast to move
 forwardX  = 600     # when to scroll
 backwardX = 230     # when to scroll
@@ -219,6 +259,8 @@
             platform.rect.x -= scroll
         for loot in loot_list:
             loot.rect.x -= scroll
+        for enemy in enemies:
+            enemy.rect.x -= scroll
 
     # scroll the world backward
     if player.rect.x <= backwardX:
@@ -228,6 +270,8 @@
             platform.rect.x += scroll
         for loot in loot_list:
             loot.rect.x += scroll
+        for enemy in enemies:
+            enemy.rect.x += scroll
                 
 #    screen.fill(black)
     screen.blit(backdrop, backdropRect)
@@ -235,6 +279,8 @@
     loot_list.draw(screen)       # refresh loot
     player.gravity()             # check gravity
     player.update(platform_list,loot_list) # refresh player
+    enemy.move()
     movingsprites.draw(screen)   # refresh player position
+    enemies.draw(screen)
     pygame.display.flip()
     clock.tick(fps)
